packages/nvidia_nat_core/src/nat/profiler/prediction_trie/trie_builder.py

```
@@ -140,41 +176,218 @@ def _extract_llm_contexts(self, steps: list[IntermediateStep]) -> list[LLMCallCo
 
     def _compute_sensitivity_scores(self, contexts: list[LLMCallContext]) -> None:
-        """Compute composite sensitivity scores for each call in the trace."""
+        """Compute composite sensitivity scores for each call in the trace.
+
+        Parallel siblings are detected via temporal overlap and assigned the
+        same logical position so that the U-shaped position signal and fan-out
+        signal treat them as a single workflow step rather than spreading them
+        across sequential indices.
+
+        After computing raw weighted scores, the values are min-max normalized
+        across all calls in the trace so the full 0–1 range is used.  This
```

Ruff (0.15.2)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)